src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -518,7 +518,17 @@ private function createEncoder($config, ContainerBuilder $container)
         // Argon2i encoder
         if ('argon2i' === $config['algorithm']) {
             if (!Argon2iPasswordEncoder::isSupported()) {
-                throw new InvalidConfigurationException('Argon2i algorithm is not supported. Please install the libsodium extension or upgrade to PHP 7.2+.');
+                // Debugging on Heroku and derivatives is a lot of work, show specific error to save end users some time
+                if (false !== strpos($_ENV['PATH'], 'heroku')) {
+                    throw new InvalidConfigurationException('Argon2i algorithm is not supported on Heroku, use BCrypt instead.');
+                }
```

Comment:
👎 for adding a check for heroku. Why heroku (and what happens if their path ever change), why not the other ones, and so on.

Reply:
Yea, we don't add such platform specific errors.

Reply:
It took me a whole day to find the issue, if this error was in here I wouldn't have to do that.

This will help a lot of devs.

Reply:
It is not the right place, doing that for all platforms/OS would be endless. Better write about it as a tweet or a blogpost, we won't add this to the codebase.

Reply:
Okay, I'll fix it up tomorrow.